appsScript/HandleFormSubmit.js

```diff
@@ -0,0 +1,215 @@
+      };
+
+      break;
+    case 'Programming Languages':
+    case 'Change Log':
+    case 'Runtime Platform':
+    case 'Status':
+      if ('schema:SoftwareSourceCode' in taxonomicService) {
+        taxonomicService['schema:SoftwareSourceCode'][ReferenceField(itemTitle)] = itemResponse;
+      };
+
+      break;
+    case 'Software License':
+      taxonomicService['schema:SoftwareSourceCode'][ReferenceField(itemTitle)] = licences.licenses.find(licence => licence.name === itemResponse).licenseId;
+
+      break;
+    case 'Content URL':
+      /* Check if index is not zero, otherwise create it first */
+      if (!associatedMediaIndex) {
+        taxonomicService['schema:AssociatedMedia'] = [];
+      }
+
+      /* Increment index by one */
+      associatedMediaIndex++;
+
+      taxonomicService['schema:AssociatedMedia'][associatedMediaIndex - 1] = {
+        [`${ReferenceField(itemTitle)}`]: itemResponse
+      };
+
+      break;
+    case 'Media License 1':
+    case 'Media License 2':
+    case 'Media License 3':
+      taxonomicService['schema:AssociatedMedia'][associatedMediaIndex - 1][`${ReferenceField(itemTitle)}`] = licences.licenses.find(licence => licence.name === itemResponse).licenseId;
+
+      break;
+    case 'Service License':
+      taxonomicService[ReferenceField(itemTitle)] = licences.licenses.find(licence => licence.name === itemResponse).licenseId;
+
+      break;
+    default:
+      taxonomicService[ReferenceField(itemTitle)] = itemResponse;
+  };
+};
+
+/**
+ * Function for handling a form submit
+ */
+const HandleFormSubmit = (e) => {
+  /* Base variables */
+  const taxonomicService = {
+    type: 'TaxonomicService',
+    attributes: {
+      content: {
+        taxonomicService: {
+          '@type': 'taxonomicService',
+          'schema:status': 'draft',
+          'schema:ratingValue': 0,
+          'schema:Service': {}
+        }
+      }
+    }
+  };
+  const itemResponses = e.response.getItemResponses();
+  let message = 'A new application has been submitted, the data:\n \n';
+
+  itemResponses.forEach((itemResponsesRecord, i) => {
+    const itemTitle = itemResponsesRecord.getItem().getTitle();
+    const itemResponse = itemResponsesRecord.getResponse();
+
+    /* Add response item to taxonomic service */
+    if (itemResponse) {
+      AddToTaxonomicService(taxonomicService.attributes.content.taxonomicService, itemTitle, itemResponse);
+    }
+
+    /* Add response item to message */
+    message += itemTitle + ': ' + itemResponse + '\n';
+  });
+
+  /* POST submission as a draft record to Cordra */
+  const url = "https://cetaf-marketplace.dissco.tech/cordra/doip/0.DOIP/Op.Create?targetId=service";
+  const options = {
+    method: "POST",
+    headers: {
+      "Authorization": "Basic " + Utilities.base64Encode("GoogleForms" + ":" + "J6ts896bNRKw")
+    },
+    contentType: 'application/json',
+    payload: JSON.stringify(taxonomicService)
+  };
+
+  UrlFetchApp.fetch(url, options);
```

Comment:
do you need an `await` here or something?

Reply:
No because the application does not need the response data, its just a post to Cordra